### ISSUE TITLE
Update UnitView to be compatible with future faf patch

### ISSUE THREAD

**Issue by Exotic-Retard** [FAForever/nomads#582]
Future faf patch is incompatible with some of the changes Nomads makes, so we need to update the code to match.

**Comment by speed2CZ** [comment]
```
WARNING: Error running HandleEvent script in CScriptObject at 590af140: ...a\documents\github\fa\lua\ui\game\unitviewdetail.lua(834): access to nonexistent global variable "GetYield"
         stack traceback:
         	[C]: in function `error'
         	...alliance\gamedata\mohodata.scd\lua\system\config.lua(53): in function <...alliance\gamedata\mohodata.scd\lua\system\config.lua:52>
         	...a\documents\github\fa\lua\ui\game\unitviewdetail.lua(834): in function `GetUpkeep'
         	...a\documents\github\fa\lua\ui\game\unitviewdetail.lua(743): in function `DisplayResources'
         	...a\documents\github\fa\lua\ui\game\unitviewdetail.lua(701): in function <...a\documents\github\fa\lua\ui\game\unitviewdetail.lua:652>
         	...a\documents\github\fa\lua\ui\game\unitviewdetail.lua(897): in function `Show'
         	...dia\documents\github\fa\lua\ui\game\construction.lua(1158): in function <...dia\documents\github\fa\lua\ui\game\construction.lua:1076>
         	...dia\documents\github\fa\lua\ui\game\construction.lua(2986): in function `OnRolloverEvent'
         	...s\multimedia\documents\github\fa\lua\maui\button.lua(62): in function <...s\multimedia\documents\github\fa\lua\maui\button.lua:50>
```
I didnt find any related change in the FA repo that would cause this.

**Comment by speed2CZ** [comment]
Actually this one :S https://github.com/FAForever/fa/commit/4be1f8a85ded0ef21e086d72bc0933426368e445#diff-231aeae30dc4931f29ab0dad2ce29bb5

**Comment by Exotic-Retard** [comment]
fixed here c89b0071668ee47b55ca88c946432415831b7da3 and c25cc3d95c607e0e2d2bb1b3f05e45ee61df8b96

Second one is alright, first one is ultramegacursed though, fuck me xD
Can get uncursed later i suppose. But yeah due to Retarded magic got it to work both now and in the future.

Good luck to any other mods that touch this i suppose.
